### ISSUE TITLE
Support for Socks/Http Proxy?

### ISSUE THREAD

**Issue by GammaPi** [vector-im/riotX-android#488]
Since synapse is in it's early stage, a built-in proxy support (no matter it's socks or http) is necessary to keep the server safe from attackers.

I'd like to contribute to this when I have some spare time. But it would also be great if some developers are willing to do this in advance.

 #1234 related

**Comment by GammaPi** [comment]
It seems that riot-android(the old app) use matrix-android-sdk to communicate with matrix server. And matrix-android-sdk uses okhttp3 to send http request. Am I correct on this?

https://github.com/matrix-org/matrix-android-sdk/search?q=import+okhttp3.OkHttpClient%3B&unscoped_q=import+okhttp3.OkHttpClient%3B

I'm wondering if I can configure proxy for okhttpclient in  matrix-sdk/src/main/java/org/matrix/androidsdk/RestClient.java and then do some UI adjustments to solve this issue?

**Comment by dklimpel** [comment]
See also: https://github.com/matrix-org/matrix-android-sdk/pull/519

**Comment by Adam-Stambaugh** [comment]
SOCKS support would pair well with Orbot. Orbot does support a VPN mode, but that's not a great solution for some configurations.

**Comment by GammaPi** [comment]
Looking forward to this

**Comment by JJohnGreenSr** [comment]
I feel the same, Gamma

**Comment by reg3xp** [comment]
SOCKS5 would be great

**Comment by seth586** [comment]
yeah as an experiment I set up a onion_address:443 routed to my syanpse_server:8008, but the clients just dont seem to want to resolve onion addresses.

Adding a simple SOCKS5 proxy to pair with orbot would be a good first step in bringing anonymity to the matrix protocol!

**Comment by cm157** [comment]
Please add socks5 / Orbot support so can access synapse server running as hidden service

**Comment by CalunVier** [comment]
Although we have many alternatives, the built-in proxy function can make our access to martix more convenient. As we all know, not all parts of the world can easily connect with the outside world.

**Comment by dvb15** [comment]
Still need SOCKS proxy settings in UI. Using VPN in android is very inconvenient.
